worker/src/lib/HeliconeProxyRequest/forwarder.ts

```diff
@@ -130,44 +173,98 @@ export async function proxyForwarder(
     }
   }
 
-  // TODO move this to proxyRequest
-  const { data: cacheSettings, error: cacheError } = getCacheSettings(
-    proxyRequest.requestWrapper.getHeaders(),
-    proxyRequest.isStream
-  );
+  if (
+    proxyRequest.requestWrapper.heliconeHeaders.moderationsEnabled &&
+    provider == "OPENAI"
+  ) {
+    let latestMessage;
 
-  if (cacheError !== null) {
-    return responseBuilder.build({
-      body: cacheError,
-      status: 500,
-    });
-  }
+    try {
+      latestMessage = JSON.parse(proxyRequest.bodyText ?? "").messages.pop();
+    } catch (error) {
+      console.error("Error parsing latest message:", error);
+      return responseBuilder.build({
+        body: "Failed to parse the latest message.",
+        status: 500,
+      });
+    }
 
-  if (cacheSettings.shouldReadFromCache) {
-    const { data: auth, error: authError } = await request.auth();
-    if (authError == null) {
-      const db = new DBWrapper(env, auth);
-      const { data: orgData, error: orgError } = await db.getAuthParams();
-      if (orgError !== null || !orgData?.organizationId) {
-        console.error("Error getting org", orgError);
-      } else {
-        const cachedResponse = await getCachedResponse(
-          proxyRequest,
-          cacheSettings.bucketSettings,
-          env.CACHE_KV,
-          cacheSettings.cacheSeed
-        );
-        if (cachedResponse) {
-          ctx.waitUntil(
-            recordCacheHit(
-              cachedResponse.headers,
-              env,
-              new ClickhouseClientWrapper(env),
-              orgData.organizationId
-            )
-          );
-          return cachedResponse;
+    if (latestMessage.role == "user") {
```

Comment:
Last message has an implicit type of any.

can you change the type of `latestMessage` to be `myType | null` and set it to null above to safely grab content?

```ts
let latestMessage: {
   user?: string,
   content?: unknown
} | null = null;
```
```suggestion
if (latestMessage !== null && latestMessage?.role == "user")
````

Reply:
Hi, any reason to have `content?: unknown` instead of `content?: string`?